account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -55,15 +56,65 @@ import type { PermitMessage, PermitDomain } from "../gas-manager.js";
  * });
  * ```
  *
- * @param {string | string[]} policyId the policyId (or list of policyIds) for Alchemy's gas manager
- * @returns {Pick<ClientMiddlewareConfig, "dummyPaymasterAndData" | "paymasterAndData">} partial client middleware configuration containing `dummyPaymasterAndData` and `paymasterAndData`
+ * @param {string | string[]} policyId - The policyId (or list of policyIds) for Alchemy's gas manager
+ * @param {PolicyToken | undefined} policyToken - The policy token configuration
+ * @returns {Pick<ClientMiddlewareConfig, "dummyPaymasterAndData" | "paymasterAndData">} Partial client middleware configuration containing `dummyPaymasterAndData` and `paymasterAndData`
  */
 export function alchemyGasManagerMiddleware(
-  policyId: string | string[]
+  policyId: string | string[],
+  policyToken: PolicyToken | undefined
 ): Pick<ClientMiddlewareConfig, "dummyPaymasterAndData" | "paymasterAndData"> {
-  return erc7677Middleware<{ policyId: string | string[] }>({
-    context: { policyId: policyId },
-  });
+  let _context: {
+    policyId: string | string[];
+    erc20Context?: {
+      tokenAddress: string;
+      maxTokenAmount?: BigInt;
+      permit?: Hex;
+    };
+  } = { policyId: policyId };
+  return {
+    dummyPaymasterAndData: async (uo, args) => {
+      const { account, client } = await resolveProperties(args);
```

Comment:
I don't think we need resolve properties here, args shouldn't have any promises in there